app/dashboard/user/[id]/tabs/DataTab.tsx

```diff
@@ -0,0 +1,140 @@
+import { Button } from '@/components/ui/button';
+import { ApiRoutes } from '@/enums/routes.enums';
+import axiosInstance from '@/lib/axiosInstance';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import ConfirmPopup from '../components/ConfirmPopup';
+import FormField from '../components/FormField';
+import { UserDataSchema } from '../validation_schema';
+
+interface FormValues {
+  name?: string;
+  last_name?: string;
+  email?: string;
+  phone_number?: string;
+  voivodship?: string;
+  city?: string;
+  zip_code?: string;
+  street?: string;
+  house_number?: string;
+}
+
+interface DataTabProps extends FormValues {
+  userID?: string;
+}
+
+const DataTab: React.FC<DataTabProps> = ({ userID, ...initialValues }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [formData, setFormData] = useState<FormValues | null>(null);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormValues>({
+    resolver: zodResolver(UserDataSchema),
+    defaultValues: initialValues,
+  });
+  const openModal = (data: FormValues) => {
+    setFormData(data);
+    setIsModalOpen(true);
+  };
+
+  const handleConfirm = async () => {
+    if (!formData || !userID) return;
+
+    try {
+      await axiosInstance.put(`${ApiRoutes.User(userID)}`, formData);
+      setIsModalOpen(false);
+      window.location.reload();
```

Comment:
It's causing toast do disappear
try reloading/redirecting based on nextjs routing if it wont work we're gonna use state management based on react-query